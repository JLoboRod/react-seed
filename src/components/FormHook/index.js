import React from "react";

import Box from "../Box";

const FormHook = () => {
  const [state, setState] = React.useState({});

  const handleChange = ({ target: { name, value } }) => {
    setState({ ...state, [name]: value });
  };

  return (
    <Box>
      <Box>
        <form>
          <input
            type="text"
            name="name"
            className="input"
            value={state.name}
            onChange={handleChange}
          />
        </form>
      </Box>
    </Box>
  );
};

export default FormHook;
