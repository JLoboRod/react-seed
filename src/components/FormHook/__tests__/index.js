import React from "react";
import { shallow } from "enzyme";

import FormHook from "../index";

describe("<FormHook />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<FormHook />);

    expect(wrapper).toMatchSnapshot();
  })

  it("handles changes", () => {
    const wrapper = shallow(<FormHook />);
    const name = "name";
    const value = "Juan";
    const event = {
      target: {
        name,
        value
      }
    };

    wrapper.find(".input").simulate("change", event);

    expect(wrapper.find(".input").prop("name")).toBe(name);
    expect(wrapper.find(".input").prop("value")).toBe(value);
  })
});