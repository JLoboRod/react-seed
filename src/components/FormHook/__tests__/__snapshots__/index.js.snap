// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHook /> renders correctly 1`] = `
<Box>
  <Box>
    <form>
      <input
        className="input"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </form>
  </Box>
</Box>
`;
